CMakeLists.txt

```diff
@@ -465,10 +472,15 @@ if (NOT MSVC)
     endif()
 endif()
 
-if (${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "aarch64")
+if (${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "aarch64" OR "${CMAKE_GENERATOR_PLATFORM_LWR}" MATCHES "arm64")
     message(STATUS "ARM detected")
     if (MSVC)
-        # TODO: arm msvc?
+        add_compile_definitions(/fp:fast)
```

Comment:
On non-Windows platforms, this must be enabled by LLAMA_FAST in the Makefile, because it may change the results of floating-point operations. See e.g. #2977. If we're going to add this to cmake, we should add it for all platforms and make it non-default - but that's not really in scope for this PR.